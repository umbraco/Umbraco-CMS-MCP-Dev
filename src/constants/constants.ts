export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
