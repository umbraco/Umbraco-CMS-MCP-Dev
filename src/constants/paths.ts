import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');