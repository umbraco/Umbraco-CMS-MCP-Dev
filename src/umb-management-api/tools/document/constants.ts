export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}