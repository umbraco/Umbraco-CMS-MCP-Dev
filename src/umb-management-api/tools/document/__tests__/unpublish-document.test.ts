import UnpublishDocumentTool from "../put/unpublish-document.js";
import PublishDocumentTool from "../put/publish-document.js";
import { DocumentBuilder } from "./helpers/document-builder.js";
import { DocumentTestHelper } from "./helpers/document-test-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";

const TEST_DOCUMENT_NAME = "_Test UnpublishDocument";

describe("unpublish-document", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await DocumentTestHelper.cleanup(TEST_DOCUMENT_NAME);
  });

  it("should unpublish a published document", async () => {
    const builder = await new DocumentBuilder()
      .withName(TEST_DOCUMENT_NAME)
      .withRootDocumentType()
      .create();
    await builder.publish();
    const item = builder.getCreatedItem();

    // Unpublish the document
    const result = await UnpublishDocumentTool().handler(
      {
        id: item.id,
        data: { cultures: null },
      },
      { signal: new AbortController().signal }
    );
    expect(result).toMatchSnapshot();

    // Get the document back and check its published state
    const found = await DocumentTestHelper.findDocument(TEST_DOCUMENT_NAME);
    expect(found).toBeDefined();
    // Check if all variants are not published
    const isPublished = found?.variants.some(
      (v: any) => v.state === "Published"
    );
    expect(isPublished).toBe(false);
  });

  it("should handle unpublishing a non-existent document", async () => {
    const result = await UnpublishDocumentTool().handler(
      {
        id: BLANK_UUID,
        data: { cultures: [] },
      },
      { signal: new AbortController().signal }
    );
    expect(result).toMatchSnapshot();
  });
});
