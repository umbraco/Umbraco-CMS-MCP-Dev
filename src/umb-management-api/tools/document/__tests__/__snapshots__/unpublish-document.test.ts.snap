// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unpublish-document should handle unpublishing a non-existent document 1`] = `
{
  "content": [
    {
      "text": "Error using unpublish-document:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "The requested document could not be found",
    "status": 404,
    "operationStatus": "ContentNotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;

exports[`unpublish-document should unpublish a published document 1`] = `
{
  "content": [
    {
      "text": """",
      "type": "text",
    },
  ],
}
`;
